workflows: Extend IPsec tests to cover egress gateway

pchaigno opened the pull request:
The first two commits are cleanup for the IPsec workflows. The third extends coverage to IPsec + KPR + egress gateway. The last one avoids the matrix deduplication and prettyfies the job display.

pchaigno committed:
workflows: Same configs between two IPsec workflows

This commit simply ensures the two IPsec workflows use the same configs.
The configs from the end-to-end test workflow, which seem more
extensive, are used.

Note that this means IPsec+KPR is now covered in up/downgrade tests. It
was only covered in end-to-end tests before. This is now okay because
IPsec+KPR should work in v1.16.

Signed-off-by: Paul Chaignon <paul.chaignon@gmail.com>

pchaigno committed:
workflows/ipsec: Remove unnecessary config fields

The 'encryption' and 'encryption-node' field are removed. We already
know this is about IPsec encryption and node encryption isn't supported,
so those fields are just unnecessary verbosity.

Signed-off-by: Paul Chaignon <paul.chaignon@gmail.com>

pchaigno committed:
workflows/ipsec: Cover egress gateway

For the end-to-end workflow, where we check for unencrypted pod-to-pod
traffic, this change requires us to run the egress gateway tests outside
of the unencrypted traffic check. Otherwise, pod-to-world traffic
redirected to the gateway is detected as incorrect unencrypted.

Signed-off-by: Paul Chaignon <paul.chaignon@gmail.com>

aanm reviewed (approved):
@pchaigno shall we backport this PR?

aanm commented on .github/workflows/tests-ipsec-upgrade.yaml:
```diff
@@ -100,95 +100,70 @@ jobs:
       max-parallel: 16
       matrix:
-        config: ['5.4', '5.10', '6.1', '5.15', 'bpf-next']
+        name: ['1', '2', '3', '4', '5', '6', '7', '8']
```
I just noticed the output of the tests results, can we give them a more descriptive name? https://github.com/cilium/cilium/actions/runs/11332423439

At least something similar as https://github.com/cilium/cilium/actions/runs/11332421340

pchaigno committed:
workflows: Generate IPsec matrix from config file

To avoid duplicating the test configurations, let's store it in a single
place, as a YAML file, and generate the matrices for the workflows from
that file.

For the end-to-end workflow, we can take the YAML file as is. For the
upgrade workflows, we need to remove unused fields and duplicate each
entry for "mode: patch" and "mode: minor".

For the upgrade workflow, this has the added benefit of displaying the
full configuration in the job's name (instead of the previous
"(1, minor)").

Signed-off-by: Paul Chaignon <paul.chaignon@gmail.com>

julianwiedmann commented on .github/actions/ipsec/configs.yaml:
```diff
@@ -0,0 +1,86 @@
+- name: '1'
+  # renovate: datasource=docker depName=quay.io/lvh-images/kind
+  kernel: 'rhel8-20240404.144247'
+  kube-proxy: 'iptables'
```
Note that this file is currently not in scope of our renovate config, so we'll no longer auto-bump the LVH images.

pchaigno replied:
Thanks! Fix is at https://github.com/cilium/cilium/pull/35555.
